### PR TITLE
CC-918 Add Kafka Connect Maven plugin to build

### PR DESCRIPTION
**Pull request by C0urante**
**IMPORTANT**: This PR is targeted at `master` for the moment but should be redirected to `4.1.x` **after** the release occurs. DO NOT MERGE UNTIL THEN

Edit: We've now decided to redirect to `4.0.x`, and will be merging after the release is officially announced.

This PR adds the Kafka Connect Maven plugin to the project's build. Running `mvn package` should cause a plugin archive file to be generated in the `target/plugins/packages` directory, named `confluentinc-kafka-connect-elasticsearch-<version>-plugin.zip`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -124,6 +124,61 @@
...
+                            <ownerLogo>logos/confluent.png</ownerLogo>
+
+                            <pluginTypes>
+                                <pluginType>sink</pluginType>
+                            </pluginTypes>
+
+                            <tags>
+                                <tag>elasticsearch</tag>
+                                <tag>search</tag>
+                                <tag>key value store</tag>
```
Comment (rhauch) [inline pom.xml]: How about adding `Elastic`, `analytics`, and `log`. I'm not sure whether `key value store`  is really an accurate descriptor for ES, so I'd suggest removing it.

File: pom.xml [hunk 2]
```diff
@@ -124,6 +124,61 @@
...
+                <version>0.9.0-SNAPSHOT</version>
+                <artifactId>kafka-connect-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>kafka-connect</goal>
+                        </goals>
+                        <configuration>
+                            <title>Kafka Connect Elasticsearch</title>
+                            <logo>logos/elasticsearch.png</logo>
```
Comment (rhauch) [inline pom.xml]: As with the other PRs, where is the documentation that says we can use this logo?

File: pom.xml [hunk 3]
```diff
@@ -124,6 +124,61 @@
...
+
+                            <tags>
+                                <tag>elasticsearch</tag>
+                                <tag>search</tag>
+                                <tag>key value store</tag>
+                            </tags>
+
+                            <requirements>
+                                <requirement>Confluent Platform installed and running</requirement>
+                                <requirement>Elasticsearch 5.x installed and running</requirement>
```
Comment (rhauch) [inline pom.xml]: This doesn't match our documentation, and why have "...installed and running" on any of these?

File: pom.xml [hunk 4]
```diff
@@ -124,6 +124,61 @@
...
+                            </pluginTypes>
+
+                            <tags>
+                                <tag>elasticsearch</tag>
+                                <tag>search</tag>
+                                <tag>key value store</tag>
+                            </tags>
+
+                            <requirements>
+                                <requirement>Confluent Platform installed and running</requirement>
```
Comment (rhauch) [inline pom.xml]: Is this necessary? Same comment as the other PRs.

File: pom.xml (hunk 3 above)
Comment (C0urante) [reply pom.xml]: Grabbed that from our [quickstart docs](https://docs.confluent.io/current/connect/connect-elasticsearch/docs/elasticsearch_connector.html#quick-start); should it be updated here to `Elasticsearch 5.x or 6.x`?

Comment (C0urante) [reply pom.xml]: Comment got buried after removing "...installed and running"; question about ES version still stands.

Comment (rhauch) [reply pom.xml]: That's been corrected in the latest code for : https://github.com/confluentinc/kafka-connect-elasticsearch/blob/master/docs/elasticsearch_connector.rst#quick-start

**Comment by C0urante** [comment]
Looked up Elastic's logo policy, it states [here|https://www.elastic.co/brand]:
> Please Do: (...) Use the product logos to represent your integration with our open source stack
They also provided logo files there to use.

**Comment by C0urante** [comment]
@rhauch it is removed.

**Pull request merged**